### PR TITLE
fix py2/py3 build drama

### LINKED ISSUE #8: Fix TypeError for checking if key exists
When trying to find if GPG key exists, an TypeError is thrown.

```
Unexpected error: builtins.TypeError
Traceback (most recent call last):
  File "/Users/addlema/src/hcvault-management/.env/lib/python3.6/site-packages/aomi/cli.py", line 442, in main
    action_runner(parser, args)
  File "/Users/addlema/src/hcvault-management/.env/lib/python3.6/site-packages/aomi/cli.py", line 428, in action_runner
    aomi.filez.freeze(args.icefile, args)
  File "/Users/addlema/src/hcvault-management/.env/lib/python3.6/site-packages/aomi/filez.py", line 97, in freeze
    ice_file = freeze_encrypt(dest_dir, zip_filename, config, opt)
  File "/Users/addlema/src/hcvault-management/.env/lib/python3.6/site-packages/aomi/filez.py", line 72, in freeze_encrypt
    pgp_keys = grok_keys(config)
  File "/Users/addlema/src/hcvault-management/.env/lib/python3.6/site-packages/aomi/filez.py", line 40, in grok_keys
    key_id = from_keybase(key[8:])
  File "/Users/addlema/src/hcvault-management/.env/lib/python3.6/site-packages/aomi/filez.py", line 27, in from_keybase
    if not has_gpg_key(fingerprint):
  File "/Users/addlema/src/hcvault-management/.env/lib/python3.6/site-packages/cryptorito/__init__.py", line 172, in has_gpg_key
    return len([key for key in lines if key.find(fingerprint) > -1]) == 1
  File "/Users/addlema/src/hcvault-management/.env/lib/python3.6/site-packages/cryptorito/__init__.py", line 172, in <listcomp>
    return len([key for key in lines if key.find(fingerprint) > -1]) == 1
TypeError: must be str, not bytes

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/addlema/src/hcvault-management/.env/bin/aomi", line 11, in <module>
    sys.exit(main())
  File "/Users/addlema/src/hcvault-management/.env/lib/python3.6/site-packages/aomi/cli.py", line 447, in main
    unhandled(uncaught, args)
  File "/Users/addlema/src/hcvault-management/.env/lib/python3.6/site-packages/aomi/error.py", line 15, in unhandled
    output("Unexpected error: %s" % name, opt)
  File "/Users/addlema/src/hcvault-management/.env/lib/python3.6/site-packages/aomi/error.py", line 22, in output
    traceback.print_exc(sys.stderr)
  File "/Users/addlema/.pyenv/versions/3.6.2/lib/python3.6/traceback.py", line 159, in print_exc
    print_exception(*sys.exc_info(), limit=limit, file=file, chain=chain)
  File "/Users/addlema/.pyenv/versions/3.6.2/lib/python3.6/traceback.py", line 100, in print_exception
    type(value), value, tb, limit=limit).format(chain=chain):
  File "/Users/addlema/.pyenv/versions/3.6.2/lib/python3.6/traceback.py", line 497, in __init__
    capture_locals=capture_locals)
  File "/Users/addlema/.pyenv/versions/3.6.2/lib/python3.6/traceback.py", line 332, in extract
    if limit >= 0:
TypeError: '>=' not supported between instances of '_io.TextIOWrapper' and 'int'
```

### PR DESCRIPTION
**Pull request by otakup0pe**
This PR cleans up how cryptorito handles differences between Python 2 and three. These differences are mostly around unicode vs string parsing (they become consolidated in Python 3). As part of this, there are three new consolidated functions (`is_py3()`, `polite_string(a_string)`, `polite_bytes(a_string)` ) which should be used when dealing with functions that expect normal or bytelist based strings.

This PR also improves how we test our GPG integration, and extends coverage a bit. I believe this also fixes #8.

There is _also_ a new feature allowing us to (optionally) pass in a GPG passphrase via API argument. This will be used by https://github.com/Autodesk/aomi/issues/84

### CODE REVIEW

File: cryptorito/cli.py [hunk 1]
```diff
@@ -53,18 +54,32 @@ def has_key(key):
...
 
-def main():
-    """My entrypoint, let me show it to you"""
+def export_key(key_id):
+    """Export a GPG key. Note this will be binary."""
+    print(cryptorito.export_gpg_key(key_id))
+    sys.exit(0)
+
+
+def do_thing():
```
Comment (ericfode) [inline cryptorito/cli.py:81]: :+1:

File: tests/integration/a_smoke.bats [hunk 2]
```diff
@@ -7,17 +7,29 @@ setup() {
     gpg_fixture
 }
 
+@test "keybase minimal" {
+    run_cryptorito 0 import_keybase "otakup0pe"
+    run_cryptorito 2 import_keybase "otakup0pe"
+    run_cryptorito 0 import_keybase "otakup0pe:077342F6"
```
Comment (ericfode) [inline tests/integration/a_smoke.bats:13]: if you clean up your keybase account one day will that break these tests? should we make a dummy account?

**Review by ericfode — APPROVED** [review]

File: tests/integration/a_smoke.bats (hunk 2 above)
Comment (otakup0pe) [reply tests/integration/a_smoke.bats:13]: It's tru hence https://github.com/Autodesk/cryptorito/issues/17

File: cryptorito/cli.py (hunk 1 above)
Comment (otakup0pe) [reply cryptorito/cli.py:81]: Yeah the `cryptorito` cli is _really_ not meant for use outside of integration tests 😈

**Pull request merged**